pkg/domain/infra/abi/play.go

```diff
@@ -964,3 +978,33 @@ func (ic *ContainerEngine) PlayKubeDown(ctx context.Context, body io.Reader, _ e
 
 	return reports, nil
 }
+
+// playKubeSecret allows users to create and store a kubernetes secret as a podman secret
+func (ic *ContainerEngine) playKubeSecret(secret *v1.Secret) (*entities.SecretCreateReport, error) {
+	r := &entities.SecretCreateReport{}
+
+	// Create the secret manager before hand
+	secretsManager, err := ic.Libpod.SecretsManager()
+	if err != nil {
+		return nil, err
+	}
+
+	// only store the data!!! if we store the entire secret marshaled,
+	// we cannot unmarshal it later. (yaml.unmarshal does not like string --> byte conversion)
+	data, err := yaml.Marshal(secret.Data)
```

Comment:
I'm concerned that we're only storing secret data, since later on we might want to support different types of kube secrets, and might want to store other kube specific metadata that exists in the original yaml. But I'm open to discussion on this

Reply:
I concur with @ashley-cui.

@cdoern can you elaborate on the unmarshal error?

Reply:
so, in https://github.com/containers/podman/blob/0b630cd765d7ea1940b3962d352cf8d0826e65ca/pkg/specgen/generate/kube/volume.go#L137-L143 if you store the entire secret (marshaled) and thne unmarshal it into a v1.secret rather than a data map like I am here, the data in a v1.secret is a map[string][]byte but `yaml.unmarshal` does not understand that if you have "name: 1234" the 1234 can be unmarshaled as bytes rather than strings so we get a type mismatch error.

Reply:
I could make my own struct which would be almost identical but `Data` and `StringData` would be the same

Reply:
Does json marshaling work with the original struct? If so, that could also be an option

Reply:
@ashley-cui that does not seem to work either, only the custom struct seems to be possible here. I think that is a good option, right?

Reply:
Sounds good to me.